### PR TITLE
Allow bootstrapping policies with special characters in Helm

### LINKED ISSUE #14355: Unable to bootstrap policies that contain special characters (: or /) from helm chart
## Expected Behavior

Using the helm provided policy provisioning is possible with policies that include characters allowed by MinIO, but not Kubernetes ConfigMap keys.

Here's a sample yaml snippet (that would go in a `values.yaml` file for the helm chart):

```yaml
policies:
  - name: foo:admins # GitHub OIDC login
    statements:
      - actions:
          - "admin:*"
      - actions:
          - "s3:*"
        resources: 
          - 'arn:aws:s3:::*'
  - name: foo/admins # Gitlab OIDC login
    statements:
      - actions:
          - "admin:*"
      - actions:
          - "s3:*"
        resources: 
          - 'arn:aws:s3:::*'
```

## Current Behavior

Policy names must abide by the Kubernetes ConfigMap naming requirements. This is because `<policy>.json` gets created as a temporary file to pass to the "create policy job".

## Possible Solution

Since we don't need these files and they're really only being used as temporary storage we can use the policy index to manage them uniquely. I was able to get going (in a backward compatible way, no changes to the values files) with this patch:

```patch
diff --git a/helm/minio/templates/_helper_create_policy.txt b/helm/minio/templates/_helper_create_policy.txt
--- a/helm/minio/templates/_helper_create_policy.txt
+++ b/helm/minio/templates/_helper_create_policy.txt
@@ -43,17 +43,19 @@ checkPolicyExists() {
   return $?
 }

-# createPolicy($name)
+# createPolicy($name, $filename)
 createPolicy () {
   NAME=$1
+  FILENAME=$2

   # Create the name if it does not exist
+  echo "Checking policy: $NAME (in /config/$FILENAME.json)"
   if ! checkPolicyExists $NAME ; then
     echo "Creating policy '$NAME'"
   else
     echo "Policy '$NAME' already exists."
   fi
-  ${MC} admin policy add myminio $NAME /config/$NAME.json
+  ${MC} admin policy add myminio $NAME /config/$FILENAME.json

 }

@@ -67,7 +69,7 @@ connectToMinio $scheme

 {{ if .Values.policies }}
 # Create the policies
-{{- range .Values.policies }}
-createPolicy {{ .name }}
+{{- range $idx, $policy := .Values.policies }}
+createPolicy {{ $policy.name }} policy_{{ $idx }}
 {{- end }}
 {{- end }}
\ No newline at end of file
diff --git a/helm/minio/templates/configmap.yaml b/helm/minio/templates/configmap.yaml
--- a/helm/minio/templates/configmap.yaml
+++ b/helm/minio/templates/configmap.yaml
@@ -15,8 +15,9 @@ data:
 {{ include (print $.Template.BasePath "/_helper_create_user.txt") . | indent 4 }}
   add-policy: |-
 {{ include (print $.Template.BasePath "/_helper_create_policy.txt") . | indent 4 }}
-{{- range .Values.policies }}
-  {{ .name }}.json: |-
+{{- range $idx, $policy := .Values.policies }}
+  # {{ $policy.name }}
+  policy_{{ $idx }}.json: |-
 {{ include (print $.Template.BasePath "/_helper_policy.tpl") . | indent 4 }}
 {{ end }}
   custom-command: |-
```

I'm willing to open a PR, unless there's something I missed and this patch isn't required (aka "user error").

NOTE: There are a few more bits I'm actually working on with the template (to make it prettier for helm diff and allow resourceless perms (which are allowed in policies but error if the resource is nil)

## Steps to Reproduce (for bugs)

1. Create a values file like so:
   ```yaml
   policies:
     - name: foo:admins # GitHub OIDC login
       statements:
         - actions:
             - "admin:*"
         - actions:
             - "s3:*"
           resources: 
             - 'arn:aws:s3:::*'
     - name: foo/admins # Gitlab OIDC login
       statements:
         - actions:
             - "admin:*"
         - actions:
             - "s3:*"
           resources: 
             - 'arn:aws:s3:::*'
   ```
2. Run `help install -f my-values.yaml minio minio/minio`
3. Receive this error:
   ```
   Error: UPGRADE FAILED: cannot patch "minio" with kind ConfigMap: ConfigMap "minio" is invalid:
   [data[foo/admins.json]: Invalid value: "foo/admins.json": a valid config key must consist of alphanumeric
   characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is
   '[-._a-zA-Z0-9]+'), data[foo:admins.json]: Invalid value: "foo:admins.json": a valid config key must consist
   of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for
   validation is '[-._a-zA-Z0-9]+')]
   ```

## Context

I'm trying to use the provided [helm chart](https://github.com/minio/minio/tree/master/helm/minio) to perform automated policy provisioning with policies that contain both `:` characters and `/` characters. This is because my upstream providers (github and gitlab) inclue these in the groups claim.

## Regression

No

## Your Environment

* Version used (`minio --version`):
* Server setup and configuration: 4x pods provisioned via Helm chart
* Operating System and version (`uname -a`): Kubernetes

### PR DESCRIPTION
**Pull request by fernferret**
## Description
MinIO is great and I'm lazy and want to use Dex in my lab to authn/authz to MinIO. I can use Dex + github/gitlab as a lightweight IdP but they don't give me a ton of control over the groups (out of the box).

MinIO supports policies named `foo:admins` and `foo/admins` but the bootstrap script fails because Kubernetes ConfigMaps cannot have keys with non-dns names in them:

```
Error: UPGRADE FAILED: cannot patch "minio" with kind ConfigMap: ConfigMap "minio" is invalid:
[data[foo/admins.json]: Invalid value: "foo/admins.json": a valid config key must consist of alphanumeric
characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is
'[-._a-zA-Z0-9]+'), data[foo:admins.json]: Invalid value: "foo:admins.json": a valid config key must consist
of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for
validation is '[-._a-zA-Z0-9]+')]
```

## Motivation and Context

My IdP (Github and Gitlab via Dex) place a `:` (github) and a `/` (gitlab) in the JWT `groups` claim which I'm using as the policy. These policies `foo:admins` and `foo/admins` work fine in MinIO, but the bootstrap fails.

## How to test this PR?

Run `helm install -f values.yaml minio minio/minio` with this as a `values.yaml` file:

```yaml
policies:
  - name: foo:admins # GitHub OIDC login
    statements:
      - actions:
          - "admin:*"
      - actions:
          - "s3:*"
        resources: 
          - 'arn:aws:s3:::*'
  - name: foo/admins # Gitlab OIDC login
    statements:
      - actions:
          - "admin:*"
      - actions:
          - "s3:*"
        resources: 
          - 'arn:aws:s3:::*'
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

Fixes #14355

### CODE REVIEW

File: helm/minio/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Multi-Cloud Object Storage
 name: minio
-version: 3.5.6
+version: 3.5.7
 appVersion: RELEASE.2022-02-18T01-50-10Z
 keywords:
```
Comment (harshavardhana) [inline helm/minio/Chart.yaml]: do not change versions that is our work @FernFerret

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: helm/minio/Chart.yaml (hunk 1 above)
Comment (fernferret) [reply helm/minio/Chart.yaml]: Sorry about that, fixed!

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**